File: Tutoriels/Java/POO/z0_Ressources/src/manager/GProcessHello.java
//===============================================
package manager;
//===============================================
public class GProcessHello extends GProcess {
    //===============================================
    private GProcessHello() {

    }
    //===============================================
    private static class GSingleton {       
        private final static GProcessHello m_instance = new GProcessHello();
    }
    //===============================================
    public static GProcessHello Instance() {
        return GSingleton.m_instance;
    }
    //===============================================
    public void run(String[] args) {
        GPrint.Instance().print("Bonjour tout le monde");
    }
    //===============================================
}
//===============================================

File: Tutoriels/Cpp/POO/z0_Ressources/src/manager/GProcessHello.cpp
//===============================================
#include "GProcessHello.h"
#include "GPrint.h"
//===============================================
GProcessHello* GProcessHello::m_instance = 0;
//===============================================
GProcessHello::GProcessHello() {

}
//===============================================
GProcessHello::~GProcessHello() {

}
//===============================================
GProcessHello* GProcessHello::Instance() {
    if(m_instance == 0) {
        m_instance = new GProcessHello;
    }
    return m_instance;
}
//===============================================
void GProcessHello::run(int argc, char **argv) {
    GPrint::Instance()->print("Bonjour tout le monde");
}
//===============================================
